infra/storage-account/application-storage-account.bicep

```diff
@@ -0,0 +1,84 @@
+@description('Required. The parameter object for the virtual network. The object must contain the name,skuName,resourceGroup and subnetPrivateEndpoints values.')
+param vnet object
+
+@description('Required. Name of your storage account. The parameter object for storageAccount.')
+param storageAccount object
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. Environment name.')
+param environment string
+
+@allowed([
+  'Storage'
+  'StorageV2'
+  'BlobStorage'
+  'FileStorage'
+  'BlockBlobStorage'
+])
+@description('Optional. Type of Storage Account to create for the storage account.')
+param kind string = 'StorageV2'
+
+@description('Optional. Indicates whether indirect CName validation is enabled. This should only be set on updates.')
+param customDomainUseSubDomainName bool = false
+
+@description('Optional. Allows you to specify the type of endpoint. Set this to AzureDNSZone to create a large number of accounts in a single subscription, which creates accounts in an Azure DNS Zone and the endpoint URL will have an alphanumeric DNS Zone identifier.')
+@allowed([
+  ''
+  'AzureDnsZone'
+  'Standard'
+])
+param dnsEndpointType string = ''
```

Comment:
Let's use the minimal number of parameters, don't think we need customDomainUseSubDomainName and dnsEndpointType also. If we need them in future, we can add more parameters to the wrapper bicep module.

Reply:
@kbabigumira  I have made the changes👍